src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -58,12 +75,28 @@ def get_platform():
     Returns:
         A string containing the device's platform identifier
     """
-    machine_info = get_machine_info()
-    if machine_info:
-        if 'onie_platform' in machine_info:
-            return machine_info['onie_platform']
-        elif 'aboot_platform' in machine_info:
-            return machine_info['aboot_platform']
+
+    # If we are running in a container, we won't have access
+    # to machine.conf, so instead we try reading the platform
+    # identifier from Config DB
+    if in_docker_container():
+        try:
+            config_db = ConfigDBConnector()
+            config_db.connect()
+
+            metadata = config_db.get_table('DEVICE_METADATA')
+
+            if 'localhost' in metadata and 'platform' in metadata['localhost']:
+                return metadata['localhost']['platform']
+        except Exception:
+            pass
+    else:
+        machine_info = get_machine_info()
```

Comment:
can we always get from config db?

Reply:
The problem is sonic-cfggen. When not passed the `-d` flag, it shouldn't access the database. I can refactor sonic-cfggen to explicitly call get_machine_info() and parse the platform string. That should eliminate this need.

Reply:
i get it now. it looks like virtual switch has hard code the platform and sku string in the start.sh. https://github.com/Azure/sonic-buildimage/blob/master/platform/vs/docker-sonic-vs/start.sh#L5

the idea is that we can use docker vs to simulate any hardware platform. therefore, in the future we can just pass any env during the vs docker start.

therefore, i think the right way to do this is to read from the env. I like your previous code, if you can remove the db part and read from env variable. that will eliminate the db access need which fit the original design.